modules/shortcodes/vimeo.php

```diff
@@ -172,6 +172,77 @@ function vimeo_shortcode( $atts ) {
 
 add_shortcode( 'vimeo', 'vimeo_shortcode' );
 
+/**
+ * Filters the Vimeo shortcode to be AMP-compatible.
+ *
+ * @param string $html The shortcode HTML.
+ * @param string $shortcode_tag The shortcode's tag (name).
+ * @param array  $attr The attributes of the shortcode.
+ * @return string The filtered HTML.
+ */
+function amp_vimeo_shortcode( $html, $shortcode_tag, $attr ) {
+	if (
+		( class_exists( 'Jetpack_AMP_Support' ) && ! Jetpack_AMP_Support::is_amp_request() )
+		||
+		'vimeo' !== $shortcode_tag
+	) {
+		return $html;
+	}
+
+	$video_id = null;
+	if ( isset( $attr[0] ) ) {
+		$video_id = jetpack_shortcode_get_vimeo_id( $attr );
+	} elseif ( isset( $attr['id'] ) ) {
+		$video_id = $attr['id'];
+	}
+
+	list( $width, $height ) = jetpack_get_amp_vimeo_dimensions( $attr );
+	return jetpack_render_amp_vimeo( compact( 'video_id', 'width', 'height' ) );
```

Comment:
I realize you may have done this because of a CodeClimate recommendation, but I'd be willing to ignore a CodeClimate warning about a too long function here if that could help us simplify things by getting rid of `jetpack_render_amp_vimeo` and returning the `amp-vimeo` tag here.

Reply:
Thanks, that's a good idea. It's applied in https://github.com/Automattic/jetpack/pull/13921/commits/cb09ba40f08edf736c752d7bc6cc06be1a30e5e4